x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/mapper/SemanticTextFieldMapper.java

```diff
@@ -1133,4 +1236,47 @@ private static boolean canMergeModelSettings(MinimalServiceSettings previous, Mi
         conflicts.addConflict("model_settings", "");
         return false;
     }
+
+    private static boolean canMergeIndexOptions(SemanticTextIndexOptions previous, SemanticTextIndexOptions current, Conflicts conflicts) {
```

Comment:
I don't think this is needed anymore since we rely on the merge of the sub-fields to detect any conflict?